tests/console/sshd.pm

```diff
@@ -8,57 +8,77 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-# Summary: Very old test to verify sshd starts up.
+# Summary: Test to verify sshd starts and accepts connections.
 #  We need this test to succeed for followup tests using ssh localhost
-# Maintainer: Stephan Kulow <coolo@suse.de>
+#  This regression test has also an interactive part (in VirtIO console)
+# Maintainer: Stephan Kulow <coolo@suse.de>, Pavel Dostál <pdostal@suse.cz>
 
+use warnings;
 use base "consoletest";
 use strict;
 use testapi;
-use utils;
+use utils qw(systemctl exec_and_insert_password);
 use version_utils qw(is_virtualization_server is_upgrade);
 
-# check if sshd works
 sub run {
     my $self = shift;
     # new user to test sshd
     my $ssh_testman        = "sshboy";
-    my $ssh_testman_passwd = "let3me2in1";
+    my $ssh_testman_passwd = $testapi::password;
 
     select_console 'root-console';
 
+    # Stop the firewall if it's available
     if (is_upgrade && check_var('ORIGIN_SYSTEM_VERSION', '11-SP4')) {
         record_info("SuSEfirewall2 not available", "bsc#1090178: SuSEfirewall2 service is not available after upgrade from SLES11 SP4 to SLES15");
     }
     else {
         systemctl 'stop ' . $self->firewall if !is_virtualization_server;
     }
-    script_run('chkconfig sshd on');
-    assert_script_run("chkconfig sshd on", 60);
-    assert_script_run("rcsshd restart",    60);    # will do nothing if it is already running
 
-    sleep 3;                                       # give the daemon some time to start
+    # Restart sshd and check it's status
+    systemctl 'restart sshd';
+    systemctl 'status sshd';
 
-    script_run('rcsshd status', 0);
-    assert_screen 'test-sshd-1';
+    # Check that the daemons listens on right addresses/ports
+    assert_script_run q(ss -pnl4 | egrep 'tcp.*LISTEN.*:22.*sshd');
+    assert_script_run q(ss -pnl6 | egrep 'tcp.*LISTEN.*:22.*sshd');
 
     # create a new user to test sshd
     my $changepwd = $ssh_testman . ":" . $ssh_testman_passwd;
-    assert_script_run("useradd -m $ssh_testman",    60);
-    assert_script_run("echo $changepwd | chpasswd", 60);
-    $self->clear_and_verify_console;
-    select_console 'user-console';
-
-    # we output the exit status, but we don't care - we want to see the echo on screen
-    # but for debugging, it's easier to check the serial file later
-    my $str = "SSH-" . time;
-    # login use new user account
-    script_run("ssh -v $ssh_testman\@localhost -t echo LOGIN_SUCCESSFUL; echo $str-\$?- > /dev/$serialdev", 0);
-    assert_screen "ssh-login", 60;
+    assert_script_run("useradd -m $ssh_testman");
+    assert_script_run("echo $changepwd | chpasswd");
+
+    select_console('root-virtio-terminal');
```

Comment:
I doubt this will work on e.g. s390x zVM and IPMI backends. Wasn't there some support added lately to just call the normal root console and it would select the right one based on if virtio is available or not?

Reply:
there is new function introduced by @pevik  `select_serial_terminal` . Which allow to do terminal selection in a smart way

Reply:
@asmorodskyi @pevik it looks nice, but shouldn't be it documented at this place https://github.com/os-autoinst/openQA/blob/master/docs/WritingTests.asciidoc#using-a-serial-terminal ?

Reply:
Hello @okurz @asmorodskyi do you think this PR will be fine on various s390x, zVM or IPMI backends now when I implemented the ` select_serial_terminal` function?

Reply:
@czerw good point. I'll prepare PR, which updates the docs.

Reply:
@czerw I tested only some backends (qemu, svirt, ipmi), the rest was reviewed by others, but not tested. But in case of bug, there is only single place to be fixed.